src/MassTransit.Abstractions/Contexts/SendContext.cs

```diff
@@ -86,5 +86,10 @@ public interface SendContext :
         /// <returns></returns>
         SendContext<T> CreateProxy<T>(T message)
             where T : class;
+
+        /// <summary>
+        /// Provides access to tags added to metrics.
+        /// </summary>
+        IMetricsTagsFeature MetricsTagsFeature { get; }
```

Comment:
I won't allow this, as it's nothing to do with Send/Consume/Publish.

MassTransit has  a GetPayload, AddOrUpdatePayload, etc. that should be used for these things to keep them under the surface area of MassTransit.